_source/_posts/2022-06-19-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,778 @@
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID="0oa7pu3434DasvWSo4x7"
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET="09328uu098u4..."
+```
+
+The Kubernetes microservice uses Spring Cloud Config to centralize configuration (this service is part of the JHipster registry). This means that you can configure the entire microservice to use Okta as an OAuth and OpenID Connect (OIDC) provider in a single place.
+
+Open `kubernetes/registry-k8s/application-configmap.yml`. Add the following at the end of the `data.application.yml` entry (before the `jhipster-registry.yml` entry), filling in the `issuer-uri`, `client-id`, and `client-secret` taken from the `.okta.env` file. 
+
+This is the Spring Cloud Config file that the JHipster registry will serve to the entire microservice network.
+
+`kubernetes/registry-k8s/application-configmap.yml`
+
+```yaml
+data:
+  application.yml: |-
+    ...
+    spring:
+      security:
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: <client-secret>
+# app specific configuration
+jhipster-registry.yml: |-
+  ...
+```
+
+You also need to configure the registry to use OAuth. In the `kubernetes/registry-k8s/jhipster-registry.yml` file, add `oauth` to the values under `SPRING_PROFILES_ACTIVE` (as shown below).
+
+`kubernetes/registry-k8s/jhipster-registry.yml`
+
+```yaml
+spec:
+  containers:
+    - name: jhipster-registry
+      env:
+        - name: SPRING_PROFILES_ACTIVE
+          value: prod,k8s,oauth2
+```
+
+## Create an Amazon Elastic Kubernetes (EKS) cluster
+
+Now you are going to create the EKS cluster on AWS using the AWS CLI.
+
+You should have the AWS CLI installed already. Use the following command to create the Kubernetes cluster for the microservice. After you run the command, go get some coffee or lunch. This command is *slooooow*.
+
+```bash
+eksctl create cluster --name okta-k8s \
+--region us-west-2 \
+--nodegroup-name okta-k8s-nodes \
+--node-type t2.xlarge \
+--nodes 2
+```
+
+**This will take several minutes to run.**
+
+**This cluster costs money as long as it is running.** If you only keep it running for the hour or so you need to run the tutorial, the costs are small. **If you forget and leave it running for a month or two, it could cost a lot more. Don't forget to delete it when you are done!**
+
+You are using a managed cluster with EC2 nodes. You are not using the newer, Fargate managed clusters because as of right now there is a compatibility problem between Fargate and Eureka. See [this google groups thread](https://groups.google.com/g/jhipster-dev/c/148uESs6vns) and [this GitHub repository](https://github.com/jussiseppala/eurekafargatesample).
```

Comment:
FYI: I was able to run with fargate profile, but I didn't use eksctl and created them using Terraform so not sure if there was a difference. I'll do some digging for my own learning purpose

Reply:
Oh, cool. It didn't work for me. Maybe the issue was fixed?